C++ Secret Messages English Translation (#270)
NuevoFoundation/workshops: content/english/secret-messages/activity-4.md

---
title: "Activity 4 - Conditions"

> date: 2022-2-23

Reviewer: This needs to be a full date format. For example:

```suggestion
date: 2022-02-23T13:24:17-07:00
```

weight: 4
draft: true
---

Very well!

We are getting closer and closer to making our secret messages. But also, let's not forget that we have to be able to decipher those messages!

> For that, we are going to enter in the console a 1, if we want to encrypt a message, or the number 2, if we want to decrypt it.

Reviewer: ```suggestion
For that, we are going to enter in the console a `1`, if we want to encrypt a message, or the number `2`, if we want to decrypt it.
```

> For this we will need the CONDITIONS.

Reviewer: ```suggestion
For this we will need **conditions**.
```

{{% notice tip %}}

### Conditions if / else
A condition in programming is an instruction or a group of instructions, which can be executed or not, based on whether a condition is true. It would look something like this: **IF**(if) this happens, execute these instructions, **IF NOT** (else), execute these other instructions.

Example

> ...

Reviewer: All ... needs to be replaced with ``` so that they appear as code blocks on the website.

> if(3 > 1){
>   cout << "3 es mayor que 1" << endl;
> }else{
>   cout << "3 no es mayor que 1" << endl;
> }
> ...
> The program will print:
> ...
> 3 es mayor que 1
> ...

Reviewer: ```suggestion
if (3 > 1) {
  cout << "3 is bigger than 1" << endl;
} else {
  cout << "3 is not bigger than 1" << endl;
}
...
The program will print:
...
3 is bigger than 1
...
```

> There we are indicating that if the number 3 is greater than 1 '(3>1)', let what is between its keys be executed '{}'. And otherwise, let what is in the block be executed as 'else'. 

Reviewer: ```suggestion
Here we are indicating that if the number 3 is greater than 1 '(3>1)', execute the code between the first set of '{}'. Otherwise, execute the code that appears between the set of '{}' after the word 'else'. 
```

2. If this is true, then the code block immediately following `if` will be executed.
3. Otherwise, the code block following `else` will be executed. 
In the example, the `else` block will not run because the `if` condition is true - 3 is indeed greater than 1.

To create conditions, we usually use **logical operators** in the parentheses `()` beside `if`. Suppose `a = 5` and `b = 3`. Then, the following are examples of logical operators we can use:

1. Less than: `a < b` (is 5 is less than 3?)
2. Less than or equal to: `a <= b` (is 5 less than or equal to 3?)
3. Greater than: `a > b` (is 5 is greater than 3?)
4. Greater than or equal to: `a >= b` (is 5 greater than or equal to 3?)

> 5. Equals: `a == b` (is 5 equal to3?)

Reviewer: ```suggestion
5. Equals: `a == b` (is 5 equal to 3?)
```
